packages/netlify-cms-backend-bitbucket/src/API.ts

```diff
@@ -0,0 +1,675 @@
+        options.commitMessage,
+        options.status || this.initialWorkflowStatus,
+      );
+    } else {
+      // mark files for deletion
+      const diffs = await this.getDifferences(branch);
+      const toDelete: DeleteEntry[] = [];
+      for (const diff of diffs) {
+        if (!files.some(file => file.path === diff.newPath)) {
+          toDelete.push({ path: diff.newPath, delete: true });
+        }
+      }
+
+      await this.uploadFiles([...files, ...toDelete], {
+        commitMessage: options.commitMessage,
+        branch,
+      });
+    }
+  }
+
+  deleteFile = (path: string, message: string) => {
+    const body = new FormData();
+    body.append('files', path);
+    body.append('branch', this.branch);
+    if (message) {
+      body.append('message', message);
+    }
+    if (this.commitAuthor) {
+      const { name, email } = this.commitAuthor;
+      body.append('author', `${name} <${email}>`);
+    }
+    return flow([unsentRequest.withMethod('POST'), unsentRequest.withBody(body), this.request])(
+      `${this.repoURL}/src`,
+    );
+  };
+
+  generateContentKey(collectionName: string, slug: string) {
+    return generateContentKey(collectionName, slug);
+  }
+
+  contentKeyFromBranch(branch: string) {
+    return branch.substring(`${CMS_BRANCH_PREFIX}/`.length);
+  }
+
+  branchFromContentKey(contentKey: string) {
+    return `${CMS_BRANCH_PREFIX}/${contentKey}`;
+  }
+
+  async isFileExists(path: string, branch: string) {
+    const fileExists = await this.readFile(path, null, { branch })
+      .then(() => true)
+      .catch(error => {
+        if (error instanceof APIError && error.status === 404) {
+          return false;
+        }
+        throw error;
+      });
+
+    return fileExists;
+  }
+
+  async getPullRequests(sourceBranch?: string) {
+    const sourceQuery = sourceBranch
+      ? `source.branch.name = "${sourceBranch}"`
+      : `source.branch.name ~ "${CMS_BRANCH_PREFIX}/"`;
+
+    const pullRequests: BitBucketPullRequests = await this.requestJSON({
+      url: `${this.repoURL}/pullrequests`,
+      params: {
+        pagelen: 50,
+        q: oneLine`
+        source.repository.full_name = "${this.repo}" 
+        AND state = "${BitBucketPullRequestState.OPEN}" 
+        AND destination.branch.name = "${this.branch}"
+        AND comment_count > 0
+        AND ${sourceQuery}
+        `,
+      },
+    });
+
+    const labels = await Promise.all(
+      pullRequests.values.map(pr => this.getPullRequestLabel(pr.id)),
+    );
+
+    return pullRequests.values.filter((_, index) => isCMSLabel(labels[index]));
+  }
+
+  async getBranchPullRequest(branch: string) {
+    const mergeRequests = await this.getPullRequests(branch);
+    if (mergeRequests.length <= 0) {
+      throw new EditorialWorkflowError('content is not under editorial workflow', true);
+    }
+
+    return mergeRequests[0];
+  }
+
+  async retrieveMetadata(contentKey: string) {
+    const [collection, slug] = contentKey.split('/');
```

Comment:
I think this will return a wrong value for `slug`, if we are dealing with sub folder collection.

Reply:
You're probably right, let me check